src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -274,7 +274,12 @@ bool ValidateUrlField(const Json& json, const std::string& field) {
   absl::string_view host;
   absl::string_view port;
   grpc_core::SplitHostPort(url->authority(), &host, &port);
-  if (absl::ConsumeSuffix(&host, ".googleapis.com")) {
+  if (absl::ConsumeSuffix(&host, ".p.googleapis.com")) {
+    if (absl::StartsWith(host, "sts-") ||
+               absl::StartsWith(host, "iamcredentials-")) {
```

Comment:
You'll need to run `./tools/distrib/clang_format_code.sh` to make sure this is formatted correctly.